app/jobs/set_catalog_record_ids_and_barcodes_csv_job.rb

```diff
@@ -2,12 +2,41 @@
 
 ##
 # Job to update/add catalog record IDs/barcodes to objects
-class SetCatalogRecordIdsAndBarcodesCsvJob < SetCatalogRecordIdsAndBarcodesJob
+class SetCatalogRecordIdsAndBarcodesCsvJob < GenericJob
   ##
   # A job that allows a user to specify a list of druids and a list of catalog record IDs to be associated with these druids
   # @param [Integer] bulk_action_id GlobalID for a BulkAction object
   # @param [Hash] params additional parameters that an Argo job may need
+  # @option params [Array] :druids required list of
+  # @option params [String] :catalog_record_ids list of catalog record IDs to be associated 1:1 with druids in order
+  # @option params [String] :use_catalog_record_ids_option option to update the catalog_record_ids
+  # @option params [String] :barcodes list of barcodes to be associated 1:1 with druids in order
+  # @option params [String] :use_barcodes_option option to update the barcodes
   # @option params [String] :csv_file CSV string
+  def perform(bulk_action_id, params)
+    super
+
+    # Catalog record IDs, barcodes are nil if not selected for use.
+    update_druids, catalog_record_ids, barcodes, refresh = params_from(params)
+
+    with_bulk_action_log do |log|
+      update_druid_count(count: update_druids.count)
+      update_druids.each_with_index do |current_druid, i|
+        cocina_object = Repository.find(current_druid)
+        args = {}
+        args[:catalog_record_ids] = Array(catalog_record_ids[i]) if catalog_record_ids
+        args[:refresh] = refresh[i] if refresh
+        args[:barcode] = barcodes[i] if barcodes && cocina_object.dro?
+        change_set = change_set_for(cocina_object)
+        if change_set.validate(args)
+          update_catalog_record_id_and_barcode(change_set, args, log) if change_set.changed?
+        else
+          log.puts("#{Time.current} Invalid #{CatalogRecordId.label}/barcode for #{cocina_object.externalIdentifier}")
+          bulk_action.increment(:druid_count_fail).save
+        end
+      end
+    end
+  end
```

Comment:
All the changes to this job and its spec came from the deleted job (and spec), so it's not new code.